### ISSUE TITLE
FuncOf returning null on apply

### ISSUE THREAD

**Issue by neonailol** [yegor256/cactoos#551]
According to readme this library advertised and null-free

https://github.com/yegor256/cactoos/blob/8586d83a7bb4a5cd813b0b9d22f0ee6419a94d85/README.md#L35

But in `FuncOf` there is 2 constructors, that returns null on calling `apply` of retuned `Func`
Is this really nececary? Maybe they do not belong in there and should be moved to smth like `FuncNlsOf` or changed to better abstractions or simple removed

https://github.com/yegor256/cactoos/blob/8586d83a7bb4a5cd813b0b9d22f0ee6419a94d85/src/main/java/org/cactoos/func/FuncOf.java#L76-L78

https://github.com/yegor256/cactoos/blob/8586d83a7bb4a5cd813b0b9d22f0ee6419a94d85/src/main/java/org/cactoos/func/FuncOf.java#L68-L70

Also tests say that `null` is expected behaviour there

https://github.com/yegor256/cactoos/blob/8586d83a7bb4a5cd813b0b9d22f0ee6419a94d85/src/test/java/org/cactoos/func/FuncOfTest.java#L53-L64

https://github.com/yegor256/cactoos/blob/8586d83a7bb4a5cd813b0b9d22f0ee6419a94d85/src/test/java/org/cactoos/func/FuncOfTest.java#L66-L75

Also i'm seeing same behaviour in constructors of `CallableOf` and `BiFuncOf`

**Comment by 0crat** [comment]
[@yegor256](http://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by llorllale** [comment]
@yegor256 I agree with this. Let us know if you differ

**Comment by llorllale** [comment]
@0crat in

**Comment by 0crat** [comment]
> @0crat in ([here](https://github.com/yegor256/cactoos/issues/551#issuecomment-383204428))

@llorllale Job [#551](https://github.com/yegor256/cactoos/issues/551) is now in scope, role is `DEV`

**Comment by 0crat** [comment]
Bug was reported, see [§29](http://www.zerocracy.com/policy.html#29): +15 point(s) just awarded to @neonailol[/z](https://www.0crat.com/u/neonailol)

**Comment by SharplEr** [comment]
@llorllale By the way, in case if all our objects are immutable then `Proc<>` have no sense. `Proc<>` can be used only for produce some side effects which are prohibited. I think it's okay to remove `Proc<>`. What do you think?

**Comment by yegor256** [comment]
@llorllale how can we fix that? I can't see any other solution...

**Comment by SharplEr** [comment]
@llorllale Okay. In case if I remove this ctor, it will affect about 15 other classes (tests included). So a big part of functionality will be lost. Do I need to save this functionality somehow?

**Comment by SharplEr** [comment]
@yegor256  I see one way, create new class:
```Java
FuncOptional<X, Y> implements Func<X,Optional<Y>>
```
Which can return `Optional.empty()` instead of `null`.

But unfortunately such code will not build:
```Java
public CallableOf(final Runnable runnable) {
    this(new FuncOptional<>(runnable));
}
```
Now I'm thinking about can I avoid this or not.

**Comment by 0crat** [comment]
@sharpler[/z](https://www.0crat.com/u/sharpler) this job was assigned to you 5days ago. It will be taken away from you soon, unless you close it, see [§8](http://www.zerocracy.com/policy.html#8). Read [this](http://www.yegor256.com/2014/04/13/no-obligations-principle.html) and [this](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html), please.

**Comment by 0pdd** [comment]
@neonailol the puzzle [#861](https://github.com/yegor256/cactoos/issues/861) is still not solved.

**Comment by SharplEr** [comment]
@neonailol PR was merged, please close the issue.

**Comment by SharplEr** [comment]
@0crat waiting
Waiting for author reaction.

**Comment by SharplEr** [comment]
@neonailol ping

**Comment by 0crat** [comment]
> @0crat waiting Waiting for author reaction. ([here](https://github.com/yegor256/cactoos/issues/551#issuecomment-388604661))

@SharplEr The impediment for [#551](https://github.com/yegor256/cactoos/issues/551) was registered successfully by @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
@elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova) please review this job completed by @sharpler[/z](https://www.0crat.com/u/sharpler), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#551](https://github.com/yegor256/cactoos/issues/551) is now out of scope

**Comment by elenavolokhova** [comment]
@SharplEr According to our Policy:
>Some of ticket messages must mention all pull requests where the problem was actually fixed;

There is no direct message with the link to corresponding PR.
Please, confirm that you will follow this rule in future.

**Comment by SharplEr** [comment]
@elenavolokhova confirm
PR was #843

**Comment by elenavolokhova** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +30 point(s) just awarded to @sharpler[/z](https://www.0crat.com/u/sharpler)

**Comment by 0crat** [comment]
Quality review completed: +8 point(s) just awarded to @elenavolokhova[/z](https://www.0crat.com/u/elenavolokhova)

**Comment by 0pdd** [comment]
@neonailol 6 puzzles [#883](https://github.com/yegor256/cactoos/issues/883), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861).

**Comment by 0pdd** [comment]
@neonailol 5 puzzles [#883](https://github.com/yegor256/cactoos/issues/883), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#888](https://github.com/yegor256/cactoos/issues/888) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#887](https://github.com/yegor256/cactoos/issues/887).

**Comment by 0pdd** [comment]
@neonailol 4 puzzles [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#888](https://github.com/yegor256/cactoos/issues/888) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#887](https://github.com/yegor256/cactoos/issues/887).

**Comment by 0pdd** [comment]
@neonailol 4 puzzles [#1040](https://github.com/yegor256/cactoos/issues/1040), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).

**Comment by 0pdd** [comment]
@neonailol 3 puzzles [#1040](https://github.com/yegor256/cactoos/issues/1040), [#884](https://github.com/yegor256/cactoos/issues/884), [#886](https://github.com/yegor256/cactoos/issues/886) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#885](https://github.com/yegor256/cactoos/issues/885), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).

**Comment by 0pdd** [comment]
@neonailol 3 puzzles [#1040](https://github.com/yegor256/cactoos/issues/1040), [#1062](https://github.com/yegor256/cactoos/issues/1062), [#884](https://github.com/yegor256/cactoos/issues/884) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).

**Comment by 0pdd** [comment]
@neonailol 2 puzzles [#1040](https://github.com/yegor256/cactoos/issues/1040), [#1062](https://github.com/yegor256/cactoos/issues/1062) are still not solved; solved: [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).

**Comment by 0pdd** [comment]
@neonailol the puzzle [#1040](https://github.com/yegor256/cactoos/issues/1040) is still not solved; solved: [#1062](https://github.com/yegor256/cactoos/issues/1062), [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).

**Comment by 0pdd** [comment]
@neonailol all 9 puzzles are solved here: [#1040](https://github.com/yegor256/cactoos/issues/1040), [#1062](https://github.com/yegor256/cactoos/issues/1062), [#861](https://github.com/yegor256/cactoos/issues/861), [#883](https://github.com/yegor256/cactoos/issues/883), [#884](https://github.com/yegor256/cactoos/issues/884), [#885](https://github.com/yegor256/cactoos/issues/885), [#886](https://github.com/yegor256/cactoos/issues/886), [#887](https://github.com/yegor256/cactoos/issues/887), [#888](https://github.com/yegor256/cactoos/issues/888).
